lib/bignum.toit

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import encoding.hex
+import binary show BIG_ENDIAN
+
+BIGNUM_ADD_ ::= 0
+BIGNUM_SUB_ ::= 1
+BIGNUM_MUL_ ::= 2
+BIGNUM_DIV_ ::= 3
+BIGNUM_MOD_ ::= 4
+
+class Bignum:
+  negative_/bool := ?
+  limbs_/ByteArray := ?
+
+  constructor .negative_/bool .limbs_/ByteArray:
+
+  constructor.from_string data/string:
+    if data[0] == '-':
+      negative_ = true
+      data = data[1..]
+    else:
+      negative_ = false
+    
+    if data.size & 0x1 != 0:
+      data = "0" + data
+    
+    limbs_ = hex.decode data
+
+  int_to_arry_ i/int -> ByteArray:
+    result := ByteArray 8
+    BIG_ENDIAN.put_int64 result 0 i
+    return result
+
+  trans_other_ other/any -> Bignum:
+    if other is int:
+      other = Bignum
+          other < 0
+          int_to_arry_ other.abs
```

Comment:
The code for negative int is untested and does not work.

Reply:
I add negative test case, and main calculation is well, the `mod` and `exp_mod` don't support `dividend` is negative.